### PR TITLE
Google doc metadata

### PR DESCRIPTION
**Pull request by jccraft-uw**
See the commit messages for details.

This PR adds a function to get Google Drive document metadata and then uses that to create the digest for the incident report manifest.

### CODE REVIEW

File: lib/id3c/cli/io/google.py [hunk 1]
```diff
@@ -81,3 +81,15 @@ def extract_document_id_from_google_url(url_str: str) -> Optional[str]:
     google_docs_matches = re.match(google_docs_pattern, url.path)
 
     return google_docs_matches["document_id"] if google_docs_matches else None
+
+def get_document_details(document_id) -> dict:
```
Comment (kairstenfay) [inline lib/id3c/cli/io/google.py:85]: Could you please add a type hint for `document_id` here?

File: lib/id3c/cli/io/google.py [hunk 2]
```diff
@@ -81,3 +81,15 @@ def extract_document_id_from_google_url(url_str: str) -> Optional[str]:
...
+    """
+    Returns a set of metadata details about a Google Drive document.
+    """
+    drive_service = discovery.build("drive", "v2")
+    metadata = drive_service.files().get(fileId=document_id).execute()
+
+    details = {'version': metadata['version'], 'etag': metadata['etag'], 'modified_date': metadata['modifiedDate'],
+        'modifiying_user': metadata['lastModifyingUserName']}
+
+    return details
```
Comment (joverlee521) [inline lib/id3c/cli/io/google.py:95]: Just curious, why rename the fields returned by Google? I feel like it's an extra layer of field names that we would need to keep track of here.

Comment (jccraft-uw) [reply lib/id3c/cli/io/google.py:95]: I wanted the fields to match our casing convention and went with modifying_user over lastModifyUserName to simplify it.

File: lib/id3c/cli/io/google.py (hunk 1 above)
Comment (jccraft-uw) [reply lib/id3c/cli/io/google.py]: Added.

**Review by joverlee521 — APPROVED** [review]

**Pull request merged**